src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/DesignTimeDirectiveTargetExtension.cs

```diff
@@ -129,7 +129,8 @@ private void WriteDesignTimeDirectiveToken(CodeRenderingContext context, DesignT
                     }
                     break;
 
-                case DirectiveTokenKind.Namespace:
+                case DirectiveTokenKind.Namespace
+                or DirectiveTokenKind.IdentifierOrExpression:
 
                     if (string.IsNullOrEmpty(node.Content))
```

Comment:
Do I understand it correctly that this `if` is taken if `node` is a complex expression? Perhaps update the comment below to reflect that.

Reply:
This whole method is only called in the identifier case, hence why we can reuse the namespace logic we were calling before.